---
title: "Summary of CyTOF dataset"
author: "Mafe Senosain"
date: "`r Sys.Date()`"
output:
    html_document:
        toc: true
        toc_float: true
---

```{r setup, include=FALSE}
library(ggplot2)
library(RColorBrewer)
library(gridExtra)
library(reshape2)
library(jcolors)
library(ComplexHeatmap)
library(tidyr)
library(dplyr)
library(magrittr)
knitr::opts_chunk$set(echo = FALSE)
source("/Users/senosam/Documents/Repositories/Research/data_analysis_cytof/R/20_ClustAnnot_functions.R")
source("/Users/senosam/Documents/Repositories/Research/data_analysis_cytof/R/30_DA_functions.R")
source("/Users/senosam/Documents/Repositories/Research/data_analysis_cytof/R/40_DE_functions.R")
source('/Users/senosam/Documents/Repositories/Research/data_analysis_rnaseq/R/30_DEGanalysis.R')
environment_set()
```

# Data preprocessing

```{r, echo=FALSE}
# Load data
load("/Users/senosam/Documents/Massion_lab/CyTOF_summary/both/cellsubtypes.RData")
# read edited CSV file as CDE_TMA36
CDE_TMA36 <- read.csv(file = '/Users/senosam/Documents/Massion_lab/CyTOF_summary/CDE_TMA36_2020FEB25_SA.csv')
y <- read.csv('/Users/senosam/Documents/Massion_lab/radiomics_summary/TMA36_CANARY_khushbu.csv')
CDE_TMA36['SILA'] <- y$SILA_S
```

```{r, echo=FALSE}
# Data pre-processing
# Changing un-identified T cells into "Other_immune"
k <- which(annot_df$subtype_B =='T_cells')
k2 <- which(annot_df$subtype_B =='NK_cells')

annot_df['subtype_B'] <- as.character(annot_df$subtype_B)
annot_df[c(k,k2), 'subtype_B'] <- 'Other_immune'
annot_df$subtype_B <- as.factor(annot_df$subtype_B)

annot_df['subtype_B4'] <- as.character(annot_df$subtype_B)
annot_df[which(annot_df$cell_type_B=='Epithelial'), 'subtype_B4'] <- 'Epithelial'
annot_df$subtype_B4 <- as.factor(annot_df$subtype_B4)
annot_df$subtype_B4 <- factor(annot_df$subtype_B4, levels = c("Epithelial", "Endothelial", "Fib_Mesenchymal", "Other_immune", "Myeloid", "Tc_cells", "Th_cells"))

# All cell/sub types B have Fib_Mesenchymal
ref <- ref[-42,] # sample 13376 has been collected twice and will be merged, only one ref is needed
grp <- 'B'
CDE_TMA36 <- CDE_TMA36[match(ref$pt_ID, CDE_TMA36$pt_ID),] # Select only pts with CyTOF data

```

```{r}
# RNA SEQ DATA
load("/Users/senosam/Documents/Massion_lab/RNASeq_summary/rnaseq.RData")
ls_preprocessed <- preprocess_rna(path_rnaseq = '/Users/senosam/Documents/Massion_lab/RNASeq_summary/rnaseq.RData', correct_batch = T, correct_gender = T)
p_all <- p_all[which(p_all$Vantage_ID %in% colnames(ls_preprocessed$vsd_mat)),]
p_all <- p_all[match(ref$pt_ID, p_all$pt_ID),]

# CLUST EIGEN GENES
clust_eigen <- data.frame(read_tsv('/Users/senosam/Documents/Massion_lab/RNASeq_summary/Results_23_Sep_20_2/Eigengenes.tsv'))
rownames(clust_eigen) <- clust_eigen$X1
clust_eigen$X1 <- NULL
clust_eigen <- t(clust_eigen)
clust_eigen <- data.frame(clust_eigen[match(p_all$Vantage_ID, rownames(clust_eigen)),])
rownames(clust_eigen) <- ref$pt_ID

# TOP VARIANT GENES
vsd_mat <- ls_preprocessed$vsd_mat
variances <- apply(vsd_mat, 1, var)
n_genes <- length(which(variances > 0.5))
#n_genes <- 200
top_genes <- data.frame(vsd_mat) %>%
   mutate(gene=rownames(.),
          symbol=ls_preprocessed$rna_all$Feature_gene_name,
          variances = variances) %>%
   arrange(desc(variances)) %>%
   dplyr::select(gene, symbol) %>%
   head(n_genes)
vsd_matTOP<- vsd_mat[top_genes$gene,]
vsd_matTOP_ENSEMBL <- vsd_matTOP
rownames(vsd_matTOP) <- top_genes$symbol

vsd_matTOP <- t(vsd_matTOP)
vsd_matTOP <- data.frame(vsd_matTOP[match(p_all$Vantage_ID, rownames(vsd_matTOP)),])
rownames(vsd_matTOP) <- ref$pt_ID

vsd_matTOP_ENSEMBL <- t(vsd_matTOP_ENSEMBL)
vsd_matTOP_ENSEMBL <- data.frame(vsd_matTOP_ENSEMBL[match(p_all$Vantage_ID, 
                                                          rownames(vsd_matTOP_ENSEMBL)),])
rownames(vsd_matTOP_ENSEMBL) <- ref$pt_ID

# DECONVOLUTION
xcell_dcv <- data.frame(t(read.delim("~/Documents/Massion_lab/RNASeq_summary/deconvolution/output/rna_only/XCELL/xCell_rnaseq_fpkm_xCell_1132060320.txt", row.names=1)))
xcell_dcv <- xcell_dcv[p_all$Vantage_ID, ]
rownames(xcell_dcv) <- ref$pt_ID

```

```{r}
# MUTATIONAL DATA
mut_dt <- read.csv(file='/Users/senosam/Documents/Massion_lab/WES_summary/summary/binaryWES.csv', row.names = 1)

# Adding Mutational data to CDE
mut_ek <- data.frame(cbind(EGFR=mut_dt$EGFR, KRAS=mut_dt$KRAS))
mut_ek['mut'] <- 'Other'
mut_ek$mut[which(mut_ek$EGFR==1)] <- 'EGFR'
mut_ek$mut[which(mut_ek$KRAS==1)] <- 'KRAS'
rownames(mut_ek) <- rownames(mut_dt)

mut_ek <- mut_ek[match(ref$pt_ID, rownames(mut_ek)),]
rownames(mut_ek) <- ref$pt_ID
CDE_TMA36['mut'] <- mut_ek$mut
```

# Abundance analysis

## All subtypes (epi clusters grouped)
```{r, echo=FALSE}
sbst <- subset(annot_df, cell_type_A != 'Epithelial')
sbst[,paste0('subtype_', grp)] <- factor(sbst[,paste0('subtype_', grp)])
prcnt_by_pt <- ClassAbundanceByPt(data=sbst, ptID_col = 'pt_ID', 
    class_col = paste0('subtype_', grp))
```

```{r, echo=FALSE}
sbst <- subset(annot_df, cell_type_A == 'Epithelial')
sbst$subtype_A <- factor(sbst$subtype_A)
prcnt_by_ptE <- ClassAbundanceByPt(data=sbst, ptID_col = 'pt_ID', 
    class_col = 'subtype_A')

prcnt_by_ptE <- cbind(Epi_123=rowSums(prcnt_by_ptE[,c(1,3,4)]), Epi_456=rowSums(prcnt_by_ptE[,5:7]), 
                     Epi_7=prcnt_by_ptE$Epithelial_7, Epi_8910=rowSums(prcnt_by_ptE[,c(2,9,10)]))

prcnt_by_pt <- cbind(prcnt_by_pt/2, prcnt_by_ptE/2)
```

```{r}
set.seed(101)
km <- kmeans(scale(prcnt_by_pt), centers = 3, iter.max = 1000)
CDE_TMA36['cluster'] <- km$cluster
```

```{r, echo=FALSE}

ha = HeatmapAnnotation(
    Cluster = as.factor(CDE_TMA36$cluster),
    #CANARY = as.factor(CDE_TMA36$CANARY),
    SILA = CDE_TMA36$SILA,
    MUT = CDE_TMA36$mut,
    Smoking = CDE_TMA36$Smoking_Status,
    Stage_S = CDE_TMA36$Stages_simplified,
    Stage =  CDE_TMA36$X8th_ed_path_stage,
    Sex =  CDE_TMA36$Gender,
    DRP = CDE_TMA36$DRP_st,
    
    simple_anno_size = unit(0.5, "cm")
)
Heatmap(t(as.matrix(scale(prcnt_by_pt))), name = "z-score",
        heatmap_legend_param = list(color_bar = "continuous"), 
        row_names_gp = gpar(fontsize = 8),
        column_names_gp = gpar(fontsize = 8),
        column_split = CDE_TMA36$cluster,
        top_annotation = ha)

```

## Correlation of Epithelial HLA-DR+ subsets and immune cells

```{r echo=FALSE, fig.width = 12, fig.height = 4}

# Scatter plots
ggpubr::ggscatter(prcnt_by_pt*100, x = 'Epi_8910', y = c("Myeloid", 'Tc_cells', 'Th_cells'),
          add = "reg.line", conf.int = TRUE, combine = TRUE,
          cor.coef = TRUE, cor.method = "spearman", add.params = list(color = 'grey'),
          xlab = "Epi_8910", ylab = 'Relative abundance (%)')

```


```{r}
x <- xcell_dcv
x['cluster_ID'] <- as.factor(CDE_TMA36$cluster)
x <- na.omit(x)
x <- reshape2::melt(x,  id.vars = c('cluster_ID'))

violin_deconv <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=cluster_ID, y=value, fill=cluster_ID)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " - XCell deconvolution"),x="pt group", y = "Enrichment score") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("1", "2"), c("2", "3"), c("1", "3")),   
                map_signif_level=TRUE)
  dp
}

xcell_ct <- colnames(xcell_dcv)
for (i in xcell_ct){
  plot(violin_deconv(x, ct = i))
}

```


```{r}
x_sila <- cbind('cluster_ID' = as.factor(CDE_TMA36$cluster), 'SILA' = CDE_TMA36$SILA)
x_sila <- data.frame(na.omit(x_sila))
x_sila <- reshape2::melt(x_sila,  id.vars = c('cluster_ID'))
x_sila$cluster_ID <- as.factor(x_sila$cluster_ID)

ggplot(x_sila, aes(x=cluster_ID, y=value, fill=cluster_ID)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0("SILA score"),x="pt group", y = "SILA score") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("1", "2"), c("2", "3"), c("1", "3")),   
                map_signif_level=TRUE)


```

```{r}
x_nsize <- cbind('cluster_ID' = as.factor(CDE_TMA36$cluster), 'SILA' = CDE_TMA36$Path_Nodule_Size_cm)
x_nsize <- data.frame(na.omit(x_nsize))
x_nsize <- reshape2::melt(x_nsize,  id.vars = c('cluster_ID'))
x_nsize$cluster_ID <- as.factor(x_nsize$cluster_ID)

ggplot(x_nsize, aes(x=cluster_ID, y=value, fill=cluster_ID)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0("Nodule Size"),x="pt group", y = "Nodule Size (cm)") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("1", "2"), c("2", "3"), c("1", "3")),   
                map_signif_level=TRUE)


```


```{r}
x_clust <- clust_eigen
x_clust['cluster_ID'] <- as.factor(CDE_TMA36$cluster)
x_clust <- na.omit(x_clust)
x_clust <- reshape2::melt(x_clust,  id.vars = c('cluster_ID'))

violin_clust <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=cluster_ID, y=value, fill=cluster_ID)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " - Clust"),x="pt group", y = "Enrichment score") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("1", "2"), c("2", "3"), c("1", "3")),   
                map_signif_level=TRUE)
  dp
}

clust_nm <- colnames(clust_eigen)
for (i in clust_nm){
  plot(violin_clust(x_clust, ct = i))
}

```


```{r}
hladr_a <- 'ENSG00000204287'
hladr_b1 <- 'ENSG00000196126'

x_HLA <- cbind('cluster_ID' = as.factor(CDE_TMA36$cluster), 
               'HLADR_A' = vsd_matTOP_ENSEMBL[,grep(hladr_a, colnames(vsd_matTOP_ENSEMBL))], 
               'HLADR_B1' = vsd_matTOP_ENSEMBL[,grep(hladr_b1, colnames(vsd_matTOP_ENSEMBL))])
x_HLA <- data.frame(na.omit(x_HLA))
x_HLA <- reshape2::melt(x_HLA,  id.vars = c('cluster_ID'))
x_HLA$cluster_ID <- as.factor(x_HLA$cluster_ID)


violin_clust <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=cluster_ID, y=value, fill=cluster_ID)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " gene expression"),x="pt group", y = "VST") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("1", "2"), c("2", "3"), c("1", "3")),   
                map_signif_level=TRUE)
  dp
}

gene <- unique(x_HLA$variable)
for (i in gene){
  plot(violin_clust(x_HLA, ct = i))
}


```

# By mutations

```{r}
cytof_mut <- cbind(prcnt_by_pt, mut=mut_ek$mut)
cytof_mut <- data.frame(na.omit(cytof_mut))
cytof_mut <- reshape2::melt(cytof_mut,  id.vars = c('mut'))

violin_clust <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=mut, y=value, fill=mut)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " (CyTOF)"),x="", y = "cell type %") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("KRAS", "EGFR")),   
                map_signif_level=TRUE)
  dp
}

ct <- unique(cytof_mut$variable)
for (i in ct){
  plot(violin_clust(cytof_mut, ct = i))
}
```


```{r}
xcell_mut <- cbind(xcell_dcv, mut=mut_ek$mut)
xcell_mut <- data.frame(na.omit(xcell_mut))
xcell_mut <- reshape2::melt(xcell_mut,  id.vars = c('mut'))

violin_clust <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=mut, y=value, fill=mut)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " (xCell)"),x="", y = "cell type %") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("KRAS", "EGFR")),   
                map_signif_level=TRUE)
  dp
}

ct <- unique(xcell_mut$variable)
for (i in ct){
  plot(violin_clust(xcell_mut, ct = i))
}
```

```{r}

clust_mut <- cbind(clust_eigen, mut=mut_ek$mut) 
clust_mut <- data.frame(na.omit(clust_mut))
clust_mut <- reshape2::melt(clust_mut,  id.vars = c('mut'))

violin_clust <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=mut, y=value, fill=mut)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " - Clust"),x="", y = "Enrichment score") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("KRAS", "EGFR")),   
                map_signif_level=TRUE)
  dp
}

clust_nm <- colnames(clust_eigen)
for (i in clust_nm){
  plot(violin_clust(clust_mut, ct = i))
}
```


```{r}

hladr_a <- 'ENSG00000204287'
hladr_b1 <- 'ENSG00000196126'

HLA_mut <- cbind(mut=mut_ek$mut, 
               'HLADR_A' = vsd_matTOP_ENSEMBL[,grep(hladr_a, colnames(vsd_matTOP_ENSEMBL))], 
               'HLADR_B1' = vsd_matTOP_ENSEMBL[,grep(hladr_b1, colnames(vsd_matTOP_ENSEMBL))])
HLA_mut <- data.frame(na.omit(HLA_mut))
HLA_mut <- reshape2::melt(HLA_mut,  id.vars = c('mut'))
HLA_mut$value <- as.numeric(HLA_mut$value)

violin_clust <- function(x, ct) {
  x_sub <- x[which(x$variable == ct),]
  dp <- ggplot(x_sub, aes(x=mut, y=value, fill=mut)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0(ct, " gene expression"),x="", y = "VST") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("KRAS", "EGFR")),   
                map_signif_level=TRUE)
  dp
}

gene <- unique(HLA_mut$variable)
for (i in gene){
  plot(violin_clust(HLA_mut, ct = i))
}
```


```{r}
sila_mut <- cbind(mut=mut_ek$mut, 'SILA' = CDE_TMA36$SILA)
sila_mut <- data.frame(na.omit(sila_mut))
sila_mut <- reshape2::melt(sila_mut,  id.vars = c('mut'))
sila_mut$value <- as.numeric(sila_mut$value)

ggplot(sila_mut, aes(x=mut, y=value, fill=mut)) + 
    geom_violin(trim=FALSE) +
    geom_boxplot(width=0.1, fill="white") +
    labs(title=paste0("SILA score"),x="", y = "SILA score") +
    scale_fill_brewer(palette="Dark2") + theme_minimal() +
    ggsignif::geom_signif(comparisons = list(c("KRAS", "EGFR")),   
                map_signif_level=TRUE)


```


## DGE analysis
